util/contentutil/fetcher.go

```diff
@@ -55,6 +55,9 @@ func (r *readerAt) ReadAt(b []byte, off int64) (int, error) {
 	var totalN int
 	for len(b) > 0 {
 		n, err := r.Reader.Read(b)
+		if err == io.EOF && n == len(b) {
```

Comment:
@tonistiigi I don't understand this. What if `n < len(b)` ?

Reply:
Then `EOF` is a regular error. This is actually workaround for `containerd` issue where some functions do not handle `EOF` properly as https://golang.org/pkg/io/#ReaderAt defines.